pubtools/_quay/merge_manifest_list.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from .utils.misc import setup_arg_parser, add_args_env_variables
+from .manifest_list_merger import ManifestListMerger
+
+LOG = logging.getLogger()
+logging.basicConfig()
+LOG.setLevel(logging.INFO)
+
+MERGE_MANIFEST_LIST_ARGS = {
+    ("--source-ref",): {
+        "help": "Source image reference. Manifest list data of this image will "
+        "overwrite destination's manifest list",
+        "required": True,
+        "type": str,
+    },
+    ("--dest-ref",): {
+        "help": "Destination image reference. Must be specified by tag. New manifest list will be "
+        "uploaded to this image reference.",
+        "required": True,
+        "type": str,
+    },
+    ("--quay-user",): {
+        "help": "Username for Quay login.",
+        "required": True,
+        "type": str,
+    },
+    ("--quay-token",): {
+        "help": "Authentication token for Quay. Can be specified by env variable AUTH_TOKEN.",
+        "required": False,
+        "type": str,
+        "env_variable": "AUTH_TOKEN",
+    },
+}
+
+
+def verify_merge_manifest_list_args(args):
+    """Verify the presence and correctness of input parameters."""
+    if "@" in args.dest_ref:
+        raise ValueError("Destination must be specified via tag, not digest")
+
+    if not args.quay_token:
+        raise ValueError("Quay token must be set")
+
+
+def merge_manifest_list_main(sysargs=None):
+    """Entrypoint for manifest list merging."""
+    parser = setup_arg_parser(MERGE_MANIFEST_LIST_ARGS)
+    if sysargs:
+        args = parser.parse_args(sysargs[1:])
+    else:
+        args = parser.parse_args()  # pragma: no cover"
+    args = add_args_env_variables(args, MERGE_MANIFEST_LIST_ARGS)
+
+    verify_merge_manifest_list_args(args)
+    merger = ManifestListMerger(
+        args.source_ref, args.dest_ref, args.quay_user, args.quay_token
```

Comment:
it's little confusing how token is changed to password

Reply:
I guess 'password' is the more correct term, since any Quay credentials could be used. Robot token is just a special-case password. At the same time, I'm assuming that this script will be primarily used with a robot account. What do you think, which way should I unify the naming?

Reply:
Okay, I chose to rename everything as 'password'

Reply:
but token from quay contains both username and password, just base64 encoded so that's why think it's not wise to interweave "token" and "password"

Reply:
You mean robot token? I don't think that's true